cfy_manager/components/postgresql_server/postgresql_server.py

```diff
@@ -1732,15 +1739,90 @@ def start(self):
             service.verify_alive(POSTGRES_SERVICE_NAME)
         logger.notice('PostgreSQL Server successfully started')
 
+    def _save_encoding_and_locale(self):
+        self.encoding = subprocess.run(
+            ['psql', '-U', 'postgres', '-c', 'SHOW SERVER_ENCODING', '-t'],
+            stdout=subprocess.PIPE, stderr=subprocess.DEVNULL
+        ).stdout.decode('utf-8').strip()
+
+        self.locale = subprocess.run(
+            ['psql', '-U', 'postgres', '-c', 'SHOW LC_CTYPE', '-t'],
+            stdout=subprocess.PIPE, stderr=subprocess.DEVNULL
+        ).stdout.decode('utf-8').strip()
+
     def stop(self, force=True):
+        self._save_encoding_and_locale()
         logger.notice('Stopping PostgreSQL Server...')
         if config[POSTGRESQL_SERVER]['cluster']['nodes']:
             service.stop('etcd')
             service.stop('patroni')
         else:
-            service.stop(POSTGRES_SERVICE_NAME)
+            try:
+                service.stop(POSTGRES_SERVICE_NAME)
+                self.upgrading_psql = False
+            except ProcessExecutionError:
+                service.stop(OLD_POSTGRES_SERVICE_NAME)
+                service.remove(OLD_POSTGRES_SERVICE_NAME)
+                self.upgrading_psql = True
```

Comment:
ah this is unfortunate, storing stuff on the instance, a bit ugly...

- can we rename the flag to something else? `psql` is the, well, cli client
- we only really need the encoding/locale (if at all?) here, not in every `stop` call

I'm thinking something like
```python
            except ProcessExecutionError:
                self.old_postgresql_settings = self._get_encoding_and_locale()  # and this is your flag
                service.stop(OLD_POSTGRES_SERVICE_NAME)
                service.remove(OLD_POSTGRES_SERVICE_NAME)
```